src/twisted/internet/defer.py

```diff
@@ -379,44 +520,44 @@ def addTimeout(self, timeout, clock, onTimeoutCancel=None):
 
         @param timeout: number of seconds to wait before timing out this
             L{Deferred}
-        @type timeout: L{int}
-
         @param clock: The object which will be used to schedule the timeout.
-        @type clock: L{twisted.internet.interfaces.IReactorTime}
-
         @param onTimeoutCancel: A callable which is called immediately after
             this L{Deferred} times out, and not if this L{Deferred} is
             otherwise cancelled before the timeout. It takes an arbitrary
             value, which is the value of this L{Deferred} at that exact point
             in time (probably a L{CancelledError} L{Failure}), and the
             C{timeout}.  The default callable (if none is provided) will
             translate a L{CancelledError} L{Failure} into a L{TimeoutError}.
-        @type onTimeoutCancel: L{callable}
 
         @return: C{self}.
-        @rtype: a L{Deferred}
 
         @since: 16.5
         """
         timedOut = [False]
 
-        def timeItOut():
+        def timeItOut() -> None:
             timedOut[0] = True
             self.cancel()
 
         delayedCall = clock.callLater(timeout, timeItOut)
 
-        def convertCancelled(value):
+        def convertCancelled(value: object) -> object:
             # if C{deferred} was timed out, call the translation function,
             # if provdied, otherwise just use L{cancelledToTimedOutError}
             if timedOut[0]:
-                toCall = onTimeoutCancel or _cancelledToTimedOutError
+                # Default value used to be None and callers may be using None
+                if onTimeoutCancel is None:
+                    toCall = _cancelledToTimedOutError  # type: ignore[unreachable]
```

Comment:
Part of why I changed this is that I mis-read the doctoring and thought that passing in `None` wasn't documented, but I see it there now.

I'm going to revert this and leave it as a future exercise to make it not an optional callable. I think that doing so properly should probably include making `_cancelledToTimedOutError` public so that one can pass it in instead of `None`.